src/primary-navigation-layout/terra-application-navigation/extensions/_Extension.jsx

```diff
@@ -45,10 +50,11 @@ const Extension = ({
 
   return (
     <div
+      id={id}
       role="button"
       tabIndex="0"
       className={cx('extension')}
-      onClick={onSelect}
+      onClick={onSelect ? () => onSelect() : undefined}
```

Comment:
This looks redundant, but it prevents us from sending down the raw event as an argument to the callback. Caught this in testing callback args.

You'll see this change a few other places too.